cmd/dockerd/grpclog.go

```diff
@@ -0,0 +1,13 @@
+package main
+
+import (
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc/grpclog"
+)
+
+// grpc's default logger is *very* noisy and uses "info" and even "warn" level logging for mostly useless messages.
+func configureGRPCLog() {
+	l := logrus.NewEntry(logrus.StandardLogger()).WithField("library", "grpc")
+	// Note: Set the grpc logger to use the same writer as the debug level logger for all but error level logs.
+	grpclog.SetLoggerV2(grpclog.NewLoggerV2(l.WriterLevel(logrus.DebugLevel), l.WriterLevel(logrus.DebugLevel), l.WriterLevel(logrus.ErrorLevel)))
```

Comment:
That's even better than what I came up with!

Reply:
Made another change after this because I found it was still outputing warning messages that are completely useless for us.
grpclog also has a verbosity level which I tried to dial all the way down and still we end up with these messages... so I'm not even sure the point of it.